net/src/ipv6/mod.rs

```diff
@@ -430,10 +453,13 @@ mod contract {
     impl TypeGenerator for Ipv6 {
         fn generate<D: Driver>(u: &mut D) -> Option<Self> {
             let mut header = Ipv6(Ipv6Header::default());
+            #[allow(unsafe_code)] // safe in test context
+            unsafe {
+                header.set_next_header(u.gen()?);
```

Comment:
Why was this not needed before, but only after the u16 change?

Reply:
also a rebase mistake